lib/vm/src/vmcontext.rs

```diff
@@ -421,14 +422,42 @@ mod test_vmtable_definition {
     }
 }
 
+/// A typesafe wrapper around the storage for a global variables.
+///
+/// # Safety
+///
+/// Accessing the different members of this union is always safe because there
+/// are no invalid values for any of the types and the whole object is
+/// initialized by VMGlobalDefinition::new().
+#[derive(Clone, Copy)]
+#[repr(C, align(16))]
+pub union VMGlobalDefinitionStorage {
+    as_i32: i32,
+    as_u32: u32,
+    as_f32: f32,
+    as_i64: i64,
+    as_u64: u64,
+    as_f64: f64,
+    as_u128: u128,
+    bytes: [u8; 16],
+}
+
+impl fmt::Debug for VMGlobalDefinitionStorage {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("VMGlobalDefinitionStorage")
+            .field("bytes", unsafe { &self.bytes })
+            .finish()
+    }
+}
+
 /// The storage for a WebAssembly global defined within the instance.
 ///
 /// TODO: Pack the globals more densely, rather than using the same size
 /// for every type.
 #[derive(Debug, Clone)]
 #[repr(C, align(16))]
 pub struct VMGlobalDefinition {
-    storage: [u8; 16],
+    storage: VMGlobalDefinitionStorage,
```

Comment:
How hard would it be to store the `Global`'s type here? That way we could panic if the requested type isn't appropriate (e.g. casting a `f32` to `i32`).

Reply:
It really doesn't belong here. The VM* structs are an API to the compiler-generated code, and the type doesn't need to include the type since it's fixed when the wasm is being compiled.

Reply:
I'm good with that.